### PR TITLE
BUG/TST Added support for empty dictionary when passed to agg for gro…

### LINKED ISSUE #48581: BUG: Inconsistency when empty inputs are passed to agg for groupby
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
>>> import pandas as pd

>>> df = pd.DataFrame(data={'a': [1, 2, 3], 'b': [4, 5, 6]})
>>> print(df.groupby("a").agg([])) # Passing an empty list as argument
Empty DataFrame
Columns: []
Index: []
>>> print(df.groupby("a").agg(())) # Passing an empty tuple as argument
Empty DataFrame
Columns: []
Index: []
>>> print(df.groupby("a").agg({})) # Passing an empty dictionary as argument
....
ValueError: No objects to concatenate
```

### Issue Description

Passing an empty `list` or `tuple` to groupby.agg returns an empty dataframe but an empty dictionary raises a ValueError.

This is part of an effort to find consistency between pandas operations with empty inputs. Issue #47959 is an attempt to start a discussion on this.

### Expected Behavior

Some Consistency

### Installed Versions

<details>
INSTALLED VERSIONS
------------------
commit           : 519fa101f26bfa89038ebb1ce2d6a23c5f2f76ad
python           : 3.9.10.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 21.3.0
Version          : Darwin Kernel Version 21.3.0: Wed Jan  5 21:37:58 PST 2022; root:xnu-8019.80.24~20/RELEASE_ARM64_T8101
machine          : arm64
processor        : arm
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : None.UTF-8

pandas           : 1.5.0.dev0+1478.g94044c8532
numpy            : 1.22.3
pytz             : 2022.1
dateutil         : 2.8.2
setuptools       : 60.9.3
pip              : 22.1.1
Cython           : 0.29.32
pytest           : 7.1.2
hypothesis       : 6.52.3
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.8.0
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 3.1.2
IPython          : 8.3.0
pandas_datareader: 0.10.0
bs4              : 4.11.1
bottleneck       : None
brotli           : None
fastparquet      : None
fsspec           : None
gcsfs            : None
matplotlib       : 3.5.2
numba            : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : 1.8.1
snappy           : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
zstandard        : None
tzdata           : None
</details>

### PR DESCRIPTION
**Pull request by eshanja1n**
…upby

This adds consistency across empty containers when passed to agg for groupby. Also added new test to check new functionality works as expected.

- [x] closes #48581
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/apply.py [hunk 1]
```diff
@@ -413,11 +413,19 @@ def agg_dict_like(self) -> DataFrame | Series:
...
-                keys=keys_to_use,
-            )
+
+            # GH 48581 - Adding consistency to empty agg types
+            if not results:
+                # return empty DataFrame if results is empty, following pattern
+                # empty list and tuple
+                from pandas import DataFrame
+
+                result = DataFrame()
```
Comment (mroeschke) [inline pandas/core/apply.py:423]: Does `type(selected_object)()` work here?

File: pandas/tests/groupby/aggregate/test_aggregate.py [hunk 2]
```diff
@@ -1443,3 +1443,12 @@ def test_agg_of_mode_list(test, constant):
     expected = expected.set_index(0)
 
     tm.assert_frame_equal(result, expected)
+
+
+def test_empty_container_consistency():
+    df = DataFrame(data={"a": [1, 2], "b": [3, 4], "c": [5, 6]})
```
Comment (rhshadrach) [inline pandas/tests/groupby/aggregate/test_aggregate.py:1449]: Can you add the GitHub Issue number as a comment in the first line

File: pandas/core/apply.py [hunk 3]
```diff
@@ -413,11 +413,19 @@ def agg_dict_like(self) -> DataFrame | Series:
...
-            result = concat(
-                {k: results[k] for k in keys_to_use},
-                axis=axis,
-                keys=keys_to_use,
-            )
+
+            # GH 48581 - Adding consistency to empty agg types
+            if not results:
+                # return empty DataFrame if results is empty, following pattern
+                # empty list and tuple
```
Comment (rhshadrach) [inline pandas/core/apply.py:420]: I think the comments here can be removed; I think it's clear that the `else` code path will fail in this case and why we're returning an empty frame.

**Comment by mroeschke** [comment]
Thanks for the pull request, but it appears to have gone stale. If interested in continuing, please merge in the main branch, address any review comments and/or failing tests, and we can reopen.

**Pull request closed without merging**